### ISSUE TITLE
Unable To Enable Writes Due To ClusterBlockException

### ISSUE THREAD

**Issue by rholloway** [elastic/elasticsearch#11593]
So while debugging a separate issue in the cluster, I issued the following command via API:

curl -XPUT 'localhost:9200/test-*/_settings' -d '
{
    "index" : {
        "blocks": {
            "write": true,
            "metadata": true
        }
    }
}'

The reasoning is unimportant and related to my stupidity in what exactly this would do, but my plan was to flip right back to false. However, this cannot be done.

{"error":"RemoteTransportException[[es-master2][inet[/<ip>:<port>]][indices:admin/settings/update]]; nested: ClusterBlockException[blocked by: [FORBIDDEN/9/index metadata (api)];]; ","status":403}

After learning a bit more of my actions, it sort of makes sense - after all I did forbid it. However, there seems to be no way to undo this change forcefully that I have discovered. If I try to set to false via localhost on master, same response. If I shutdown all master nodes, and add the following to elasticsearch.yml, no luck:

index.blocks.metadata: false
index.blocks.write: false

I'm assuming because API + index specific settings override yml.

How can this be undone? If I could simply delete these indices, I would. But delete fails because of the same reason. I cannot ignore either, because simple things like localhost:9200/_cat/shards also fails (why, I'm not sure, maybe because index.blocks.metadata is set to true?). While these particular indexes are not so important, other more important ones are on the cluster...

So, how does one sudo or force their way out of this? There must be a way.

**Comment by rholloway** [comment]
Edit, this has been solved! The issue for me was I was trying to revert the change with

curl -XPUT 'localhost:9200/test-*/_settings' -d '
{
    "index" : {
        "blocks": {
            "write": false,
            "metadata": false
        }
    }
}'

which appears to attempt ta change the write block first, which tosses the metadata exception. If I break it into two separate queries, then I was able to revert. Lesson learned - don't lose control changing various settings you aren't familiar with while trying to fix an issue.

curl -XPUT 'localhost:9200/test-*/_settings' -d '
{
    "index" : {
        "blocks": {
            "metadata": false
        }
    }
}'

curl -XPUT 'localhost:9200/test-*/_settings' -d '
{
    "index" : {
        "blocks": {
            "write": false        
        }
    }
}'

**Comment by rholloway** [comment]
Not sure if it's worth reopening issue for, but if anyone ever does see this and happens to know where this setting is persisted to disk, I'd definitely be curious to know. I spent a while trying to track down and was going to shutdown cluster and edit manually, but it's not persisted in the global cluster state file where I thought it might be.

**Comment by clintongormley** [comment]
Hi @rholloway 

I think this is at least worth discussing.  Reopening

**Comment by clintongormley** [comment]
With https://github.com/elastic/elasticsearch/issues/15278 this setting can now be unset.
